src/DocScan/Session/Create/Filters/RequiredDocument.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yoti\DocScan\Session\Create\Filters;
+
+use Yoti\Util\Json;
+
+abstract class RequiredDocument implements \JsonSerializable
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var RequiredDocumentFilter|null
+     */
+    private $filter;
+
+    /**
+     * @param string $type
+     * @param RequiredDocumentFilter|null $filter
+     */
+    public function __construct(string $type, ?RequiredDocumentFilter $filter)
+    {
+        $this->type = $type;
+        $this->filter = $filter;
+    }
+
+    /**
+     * @return \stdClass
+     */
+    public function jsonSerialize(): \stdClass
+    {
+        return (object) Json::withoutNullValues([
+            'type' => $this->type,
+            'filter' => $this->filter,
```

Comment:
I've spoken with Mike, and we're going to move the filter onto the RequiredIdentityDocument interface.  This is due to the fact that they know all RequiredDocuments will have at least a `type`, however they don't know if they will always have a filter (e.g. for utility bills etc).

Reply:
Fixed in 2059a79